docs/DeveloperGuide.md

```diff
@@ -154,6 +154,48 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Deadline feature
+#### Implementation
+The `deadline` mechanism borrows from the current `edit` mechanism. The main idea of the mechanism is creating a new `Person` object with added `deadline`, then replace existing `Person` in list with the new `Person`.
+
+Step 1. The user is currently on the screen displaying the list of `Person`.
+
+Diagram below represents the list of `Person`
+<img src="images/DeadlineState1.png" width="574" />
+
+Step 2. The user executes `deadline 1 /d return book 1/1/2023` command to add the deadline `1/1/2023` with description `return book` to the first person in the address book.
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** Adding multiple `deadline` is supported, for example: `deadline 1 /d return book 1/1/2023 /d write report 5/1/2023`.
+
+Step 3. The `DeadlineCommand#execute()` method will find the `Person` at the index specified.
+<img src="images/DeadlineState2.png" width="574" />
+
+Step 4. After getting `personToAddDeadline`, `DeadlineCommand#execute()` method creates a new `Person` called `editedPerson` with added `Deadline` and all the other attributes of `personToAddDeadline`.
+<img src="images/DeadlineState3.png" width="574" />
+
+Step 5. The `editedPerson` replaces the `personToAddDeadline` in the list.
+<img src="images/DeadlineState4.png" width="574" />
+
+Step 6. The `UI` displays the updated list.
+
+Diagram below shows the execution of `deadline 1 /d return book 1/1/2023` command
+<img src="images/DeadlineSequenceDiagram.png" width="574" />
```

Comment:
Similar to what I suggested to @glennljw , I think it would be clearer if you clarified that PlantUML doesn't support deleting the lifeline after the destroy marker!

Reply:
Okay can do